### PR TITLE
Add Reviewer as member to an engagement

### PR DESCRIPTION
**Pull request by jadmsaadaot**
*Description of changes:*

- Add Reviewer membership type and group
- Add separate resource to fetch single user details
- Refactor validation for user group

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the met-public license (Apache 2.0).

### CODE REVIEW

**Review by VineetBala-AOT — APPROVED** [review]

**Review by djnunez-aot — APPROVED** [review]

File: met-api/src/met_api/models/staff_user.py [hunk 1]
```diff
@@ -60,6 +60,11 @@ def get_user_by_external_id(cls, _external_id) -> StaffUser:
         """Get a user with the provided external id."""
         return cls.query.filter(func.lower(StaffUser.external_id) == func.lower(_external_id)).first()
 
+    @classmethod
+    def get_user_by_id(cls, _user_id) -> StaffUser:
+        """Get a user with the provided external id."""
+        return cls.query.filter(StaffUser.id == _user_id).first()
+
```
Comment (saravanpa-aot) [inline met-api/src/met_api/models/staff_user.py:67]: if it inherits from basemodel , get_by_id method shud also be inherited..this wont be needed

File: met-api/src/met_api/services/membership_service.py [hunk 2]
```diff
@@ -17,44 +19,77 @@ class MembershipService:
...
-        existing_membership = MembershipModel.find_by_engagement_and_user_id(engagement_id, user.id)
-        if existing_membership:
+    def _get_membership_details(user: StaffUserModel):
+        group_name = Groups.EAO_TEAM_MEMBER.name
+        membership_type = MembershipType.TEAM_MEMBER
+
+        if Groups.EAO_REVIEWER.value in user.get('groups'):
+            group_name = Groups.EAO_REVIEWER.name
+            membership_type = MembershipType.REVIEWER
+        return group_name, membership_type
```
Comment (saravanpa-aot) [inline met-api/src/met_api/services/membership_service.py:48]: can you write some comments here?
not very evident on the logic!
the method name is _get_membership_details ;but feels like we might need to explain the logic in the if statement..
would be helpful to know whats the method is doing

File: met-api/src/met_api/services/membership_service.py [hunk 3]
```diff
@@ -17,44 +19,77 @@ class MembershipService:
     def create_membership(engagement_id, request_json: dict):
         """Create membership."""
         user_id = request_json.get('user_id')
-        user: StaffUserModel = StaffUserModel.get_user_by_external_id(user_id)
-        if not user:
+        user_db: StaffUserModel = StaffUserModel.get_user_by_external_id(user_id)
```
Comment (saravanpa-aot) [inline met-api/src/met_api/services/membership_service.py:22]: lol..i like the other way.. this is user ;not user_db
the other guy which gets serialised , is something like user_data or something of that sort..
he he..dont change it;but persoanl preference

File: met-api/src/met_api/services/membership_service.py (hunk 2 above)
Comment (saravanpa-aot) [reply met-api/src/met_api/services/membership_service.py:48]: i came back to this method and re-read..i think it really needs an explanation ..
sorry to be picky on this one

File: met-api/src/met_api/models/staff_user.py (hunk 1 above)
Comment (jadmsaadaot) [reply met-api/src/met_api/models/staff_user.py:67]: Thanks, removed

File: met-api/src/met_api/services/membership_service.py (hunk 2 above)
Comment (jadmsaadaot) [reply met-api/src/met_api/services/membership_service.py]: Great, thank you. Refactored can you please recheck

File: met-api/src/met_api/services/membership_service.py (hunk 3 above)
Comment (jadmsaadaot) [reply met-api/src/met_api/services/membership_service.py]: reverted the change

**Pull request merged**